platform/jobs/RetirementJobs.groovy

```diff
@@ -620,6 +620,139 @@ job('retirement-partner-report-cleanup') {
     }
 }
 
+// ########### user-retirement-archiver ###########
+// Defines the job for archiving retired user statuses
+job('user-retirement-archiver') {
+    description('Archive the old user retirement status rows to AWS S3.')
+
+    authorization {
+        blocksInheritance(true)
+        // Only core teams have full control of the archiver.
+        List membersWithFullControl = ['edx*platform-team', 'edx*testeng', 'edx*devops']
+        membersWithFullControl.each { emp ->
+            permissionAll(emp)
+        }
+        // Other engineering teams can view.
+        List extraMembersCanView = ['edx*learner']
+        extraMembersCanView.each { emp ->
+            permission('hudson.model.Item.Read', emp)
+            permission('hudson.model.Item.Discover', emp)
+        }
+    }
+
+    // This label is configured to disallow more than one simultaneous job
+    // per instance, and generally there are always available jenkins-worker
+    // instances idling.
+    label('jenkins-worker')
+
+    // Only one of these jobs should be running at a time per environment
+    concurrentBuild(true)
+    throttleConcurrentBuilds {
+        // A maxTotal of 0 implies unlimited simultaneous jobs, but below we
+        // restrict one build per environment.
+        maxTotal(0)
+    }
+    configure { project ->
+        project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+            'paramsToUseForLimit'('ENVIRONMENT')
+        project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+            'limitOneJobWithMatchingParams'('true')
+    }
+
+    // keep jobs around for 30 days
+    logRotator JENKINS_PUBLIC_LOG_ROTATOR(30)
+
+    wrappers {
+        buildUserVars() /* gives us access to BUILD_USER_ID, among other things */
+        buildName('#${BUILD_NUMBER}, ${ENV,var="ENVIRONMENT"}')
+        timestamps()
+        colorizeOutput('xterm')
+        credentialsBinding {
+            file('USER_RETIREMENT_SECURE_DEFAULT', 'user-retirement-secure-default.yml')
+        }
+    }
+
+    parameters {
+        stringParam('TUBULAR_BRANCH', 'master', 'Repo branch for the tubular scripts.')
+        stringParam('ENVIRONMENT', 'secure-default', 'edx environment from which to archive users, in ENVIRONMENT-DEPLOYMENT format.')
+        stringParam('COOL_OFF_DAYS', '60', 'Number of days after which a retired learner status should be archived in S3.')
```

Comment:
Just to confirm that this is the expectation, this would be 60 days total since the retirement was requested, not 60 days after the retirement was completed (since we have the additional cool off days before the retirement is run).

Reply:
Good point. I'll change to value to 67 and the comment to reflect this point.

Reply:
Done.